### ISSUE TITLE
fwupd without consolekit/systemd ?

### ISSUE THREAD

**Issue by joakim-tjernlund** [fwupd/fwupd#1408]
Is there any way one could build and use fwupd without consolekit/systemd ?
Our use is a PXE booted linux installer in which I would like to manually run fwupd
to update BIOS etc. before installing out Linux img
This installer does not have systemd/consolekit as it supposed to be small and lean.

**Comment by hughsie** [comment]
I think in theory you can compile without systemd and consolekit, although I don't think anyone has tested it and it's not being run as a CI target. Could you give it a go?

**Comment by superm1** [comment]
Which plugin will you be updating the BIOS with?  Keep in mind the way the UEFI plugin works it will need to have a stateful location like an EFI system partition to the store the payload for access from a different EFI bootloader (fwupdx64.efi).

You will also need to then PXE boot fwupdx64.efi after you have staged the files.

**Comment by joakim-tjernlund** [comment]
> I think in theory you can compile without systemd and consolekit, although I don't think anyone has tested it and it's not being run as a CI target. Could you give it a go?

Did a quick try and got suck on vala, libgudev, dev-libs/libgusb etc.
Seems like I have to install quite a lot more ...  also fwupd  wants polkit too? polkit wants systemd ...

**Comment by joakim-tjernlund** [comment]
> Which plugin will you be updating the BIOS with? Keep in mind the way the UEFI plugin works it will need to have a stateful location like an EFI system partition to the store the payload for access from a different EFI bootloader (fwupdx64.efi).
> 
> You will also need to then PXE boot fwupdx64.efi after you have staged the files.

I have PXE boot with both BIOS and UEFI. Any user needing new  FW will have to run
UEFI. But I didn't think of the EFI system partition so maybe this idea isn't going to work.

**Comment by superm1** [comment]
> For your purpose you should not use fwupd daemon, fwupdtool is plenty for you. It doesn't need polkit.

Specifically `-Dbuild=standalone`

**Comment by superm1** [comment]
I've got another idea for you that might help with the stateful partition problem. This is an installer you say right? So it will be making an ESP for you?

You can just mount it in the right place after install is "done" and use fwupdtool at that point. The first boot after you're all done will be a firmware update then before you boot into the actual OS you installed.

**Comment by joakim-tjernlund** [comment]
These are good ideas, need to spend some time to figure out what to do.
Thanks!

**Comment by superm1** [comment]
Cool.  If you come up with something workable, feel free to post something to the wiki about it for others.

**Comment by ckrzen** [comment]
Success! No ``systemd`` or ``elogind`` using a ``build=all``[default] of ``fwupd`` on CRUX Linux 3.5.

```
# Description: A simple daemon to allow session software to update firmware
# URL: https://github.com/fwupd/fwupd
# Maintainer:
# Packager: Chris Rainey, ckrzen at tuta dot io
# Depends on: valgrind libxmlb git libsoup vala help2man bash_completion tpm2-tss libsmbios libgusb gcab autoconf-archive

name=fwupd
version=1.3.7
release=1
source=(https://github.com/$name/$name/archive/$version.tar.gz)

build() 
{
        cd $name-$version

                meson build \
                -Dlibexecdir=/usr/lib/ \
                -Dprefix=/usr/ \
                -Ddocs=false \
                -Dnls=false \
                -Dgtkdoc=false \
                -Dstemmer=true \
                -Dvapi=false \
                -Dconsolekit=false \
                -Dsystemd=false \
                -Dgcab:docs=false \
                -Dgcab:vapi=false \
                -Dgusb:docs=false \
                -Dgusb:vapi=false

                ninja -C build

                DESTDIR=$PKG ninja -C build install
}
```

![2020-02-07-150228_1366x768_scrot](https://user-images.githubusercontent.com/6942277/74066119-1ac4b600-49bc-11ea-8599-55687b0fdfd6.png)

**Comment by jedahan** [comment]
does this still work on 1.4.6?

**Comment by superm1** [comment]
I don't see why it wouldn't.  If you're having problems, please open up a new issue with all the details.
